File: 09.math1/1193_fraction.py
#!/usr/bin/env python

n = int(input())

cnt = 1
stop = False

for i in range(1,n+1):
    for k in range(1,i+1):
        if cnt == n:
            if i%2 == 0:
                print("{}/{}".format(k, i-k+1))
            else
                print("{}/{}".format(i-k+1, k))
            stop = True
            break
        cnt += 1
    if stop: break

File: 09.math1/1193_fraction.cpp
#include <iostream>

using namespace std;

int main() {
    int n;
    int cnt;

    cin >> n;
    cnt = 1;
    for (int i=1; i<=n; ++i) {
        for (int k=1; k<=i; ++k) {
            if (cnt == n) {
                if (i%2 == 0)
                    cout << k << "/" << i-k+1 << endl;
                else
                    cout << i-k+1 << "/" << k << endl;
                return 0;
            }
            cnt++;
        }
    }
}
